### PR TITLE
Accessibility: Include visible initials in name displayed on account menu button (closes #21942)

### LINKED ISSUE #21942: Accessibility - Backoffice - Account menu label in name
### Which Umbraco version are you using?

17.2

### Bug summary

While doing a simple WCAG Audit of the backoffice I noticed some bugs.

The button to trigger the account menu shows your initials. These initials are not available in the aria-label on the button that's overwriting the visible content. Therefor the buttons is not accessible for users using speach to trigger actions.

Image

### Specifics

_No response_

### Steps to reproduce

Sign in to the backoffice

### Expected result / actual result

The aria-label contains the visible text to make sure it passes WCAG 2.5.3 (Label in name)

### PR DESCRIPTION
**Pull request by andreaslborg**
## Description
This PR addresses #21942, which notes that the account menu button violates WCAG 2.5.3 Label in Name compliance.

Previously, the button displayed user initials visually (e.g., "AB") but the accessible name only showed "Profile options", preventing voice navigation software from recognizing the button by its visible initials.

The fix ensures the accessible label includes the user's full name and initials: "User profile for Andreas Lykke Borg (AB)"
image

## Changes
- Added `getInitials()` utility function to extract first and last initial from user names
- Added test for `getInitials()` function covering normal names, edge cases, and null/undefined inputs
- Updated `current-user-header-app` component to include user name and initials in the button's accessible label
- Updated `profileOptions` localization term in all 15 language files to support name and initial placeholders
- Added fallback `profileOptionsDefault` localization key for cases when user data is not yet available

## Testing
Unit tests for `getInitials()` function validate all scenarios including null, undefined, empty strings, and numeric strings
Using a screen reader or voice navigation software, the button should be announced with its accessible label including the user's full name and initials (e.g., "User profile for Andreas Lykke Borg (AB)")
When user data is not available during component initialization, the button falls back to the localized "User profile" label

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/current-user-header-app.element.ts [hunk 1]
```diff
@@ -38,11 +39,16 @@ export class UmbCurrentUserHeaderAppElement extends UmbHeaderAppButtonElement {
 	}
 
 	override render() {
+		const userInitials = this._currentUser?.name ? getInitials(this._currentUser.name) : '';
+		const accessibleLabel = this._currentUser?.name
+			? this.localize.term('visuallyHiddenTexts_profileOptions', this._currentUser.name, userInitials)
+			: this.localize.term('visuallyHiddenTexts_profileOptions');
```
Comment (AndyButland) [inline src/Umbraco.Web.UI.Client/src/packages/user/current-user/current-user-header-app.element.ts]: Looks like there might be a slight problem here.  If `this._currentUser` is null, we'll get the literal `User profile for %0% (%1%)` rendered, as no arguments are being passed.  Perhaps we need a no arguments alternative key that just rendered "User profile".  In practice I imagine it won't be used, but given at least in code `this._currentUser` can be null, we should account for that case.

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts
Comment (AndyButland) [inline src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: Nice idea to extract this into a utility, but having done so, one of the benefits is that it should be fairly straightforward to have some front-end unit tests for it.  You'll see patterns for this on other utility classes in the code base.  Could you look to add please?

Comment (AndyButland) [inline src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: There's a slight downside in having this utility though, in that we now have the calculation of the initials both here and in the [UUI avatar component](https://github.com/umbraco/Umbraco.UI/blob/main/src/components/avatar/avatar.element.ts#L67-L99).

It would be nice to re-use that, otherwise there's small risk that we end up with different initials displayed in the component and here.  It's not exposed though, so I don't see you can really do that.  I see the implementation is different, but looks to me that what you have here is simpler and meets the needs better.

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿/**
+ * Extracts the first and last initial from a user's name
+ * @param name - The user's full name
+ * @returns The initials in uppercase
+ */
+export function getInitials(name: string): string {
+	if (!name || typeof name !== 'string') {
```
Comment (AndyButland) [inline src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts]:
```suggestion
	if (!name) {
```

I think this will suffice here, given the typed `name` parameter.

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts
Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: I had the same thought when implementing it. What would you recommend we do for now?

Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: I think keep it as it is for now - maybe just add a comment acknowledging this duplication.

Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: Alright. I'm testing right now, and I can see the user name is mandatory, however its possible to just write "1". The UUI avatar writes its as initial "1".
image

In this case, shouldn't we write 'User profile for 1 (1)'?

And same goes for a case with multiple numbers spaced out:
image

'User profile for 1 2 3 4 5 6 7 8 (18)'

Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: I agree and I have added one now. 😊 `src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.test.ts`

Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: Okay so the examples above work as expected and is also tested in the new test file. Here is a list:

```
Andreas Lykke Borg: AB
Rick Doesburg: RD
john doe: JD
Jane  Mary  Smith: JS
Madonna: M
null: 
undefined: 
empty string: 
whitespace only: 
1: 1
1 2 3: 13
```

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/current-user-header-app.element.ts (hunk 1 above)
Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/current-user-header-app.element.ts]: I have added a fallback text `profileOptionsDefault` for this case. 😊

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: Maybe then it's best to replicate the logic from the avatar, just to avoid weird edge cases where they don't align.  There's some code in there about handling emojis though - I don't suppose we need that here.

Comment (andreaslborg) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: @AndyButland I replicated it now without the emojis. 😊

**Comment by AndyButland** [comment]
@iOvergaard @nielslyngsoe - could one of you take a view please on the discussion above about having to replicate/re-apply the "get initials" logic that we already have in the avatar component in the UUI logic.  There's a theoretical chance that these diverge and thus the initials on the avatar won't match that on the label.  

I think we could just live with this duplication and slight risk.  But maybe you can see a way we could expose the logic from the UUI library so we ensure the same code is used?

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts
Comment (AndyButland) [reply src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:1]: Looks good to me.  In my testing I'm seeing an `aria-label` of "User profile for Andy Butland (AB)" which is as expected.

**Comment by andreaslborg** [comment]
> @iOvergaard @nielslyngsoe - could one of you take a view please on the discussion above about having to replicate/re-apply the "get initials" logic that we already have in the avatar component in the UUI logic. There's a theoretical chance that these diverge and thus the initials on the avatar won't match that on the label.
> 
> I think we could just live with this duplication and slight risk. But maybe you can see a way we could expose the logic from the UUI library so we ensure the same code is used?

@iOvergaard @nielslyngsoe have you had time to have a look at this? 😊

**Comment by andreaslborg** [comment]
@AndyButland Did you have a chance to discuss this with your colleagues? 😊

File: src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * Extracts the first and last initial from a user's name.
+ * Replicates logic from UUI avatar component to ensure consistency.
+ * @param name - The user's full name
+ * @returns The initials in uppercase
+ *
+ * @note This function duplicates similar logic from the UUI avatar component.
+ * It filters out parts beginning with special characters or punctuation to handle
+ * cases like "John Doe (Admin)" correctly, resulting in "JD" instead of "J(".
+ */
+export function getInitials(name: string): string {
```
Comment (AndyButland) [inline src/Umbraco.Web.UI.Client/src/packages/user/current-user/utils/get-initials.function.ts:12]:
```suggestion
// TODO: When the utility to extract initials is exposed from the UUI library, use it and deprecate this function.
export function getInitials(name: string): string {
```

**Comment by AndyButland** [comment]
Hi @andreaslborg - we just discussed.  Consensus is that we are happy with the duplication for now, but we'll add a `TODO` to replace the function with an exposed version from the UUI library.  I've added and will apply a suggestion for that.

Thanks for the contribution here.
